### DISCUSSION TITLE
Persistent anonymous user?

### DISCUSSION THREAD

**Discussion by stefan-girlich** [nhost/nhost#3001] [Q&A]
I am using `signInAnonymous()` from `useSignInAnonymous()` to sign in a user anonymously and forward them to a subsequent page (using Next.js, without a full page reload).
`useSignInAnonymous()` correctly returns the signed in anonymous `user`.

When I refresh this page, `user` is null and the call `https://local.auth.nhost.run/v1/token` results in HTTP 401 with this response body:
```json
{
    "error": "unverified-user",
    "message":"User is not verified.",
    "status":401
}
```

I can prevent this by manually assigning and verifying an email address for the anonymous user in the dashboard.

Is there a way to let anonymous users use persistent sessions or disabling verification in general?

**Comment by stefan-girlich** [comment]
I found the solution in `nhost.toml`:
```diff
-emailVerificationRequired = true
+emailVerificationRequired = false
```

**Reply by dbarrosop** [reply]
would you mind opening a github issue? might be worth looking into not requiring email verification on anonymous users if they don't have an email associated
